cpp/ql/test/library-tests/valuenumbering/GlobalValueNumbering/test.cpp

```diff
@@ -156,4 +156,27 @@ void test_read_global_different(int n) {
   global_a->y = n;
 
   int d = global_a->x;
+}
+
+// crude mock for function call testing
+namespace std {
+  class vector {
+    int m_size;
+    int data;
+  public:
+    int size() const {
+      return m_size;
+    }
+
+    void set(int x) {
+      data = x;
+    }
+  };
+}
+
+void test_vector_size(std::vector v) {
+  int a = v.size();
+  int b = v.size();
+  v.set(10);
+  int c = v.size();
```

Comment:
I'm not exactly sure how it works, but I'm impressed we match the result of `size()` on line 178 to line 179, but _not_ 181.

How easy would it be to fool it, by doing something subtler that changes the number of elements in `v`?

Reply:
Sounds like the right call to me.